test/allow-extended-operators.test.js

```diff
@@ -59,4 +46,207 @@ describe('Model.settings.allowExtendedOperators', () => {
       callback(null, [instanceFound]);
     }
   }
+
+  describe('connector.settings.allowExtendedOperators', () => {
+    context('DAO.find()', () => {
+      it('converts extended operators to string value by default', () => {
+        const TestModel = createTestModel();
+        return TestModel.find(extendedQuery()).then((results) => {
+          should(results[0].value).eql('[object Object]');
+        });
+      });
+
+      it('preserves extended operators with allowExtendedOperators set', () => {
+        const TestModel = createTestModel({allowExtendedOperators: true});
+        return TestModel.find(extendedQuery()).then((results) => {
+          should(results[0].value).eql({$exists: true});
+        });
+      });
+
+      function extendedQuery() {
+        // datasource modifies the query,
+        // we have to build a new object for each test
+        return {where: {value: {$exists: true}}};
+      }
+    });
+
+    context('DAO.updateAttributes()', () => {
+      it('`options.allowExtendedOperators` override connector settings - disable strict check', () => {
+        const TestModel = createTestModel({allowExtendedOperators: false}, {strict: true});
+        return TestModel.create({value: 'test'}).then(function(instance) {
+          var updateData = {$set: {value: 'changed'}};
+          var ctxData;
+          TestModel.observe('persist', function(ctx, next) {
+            ctxData = ctx.data;
+            next();
+          });
```

Comment:
This code is repeated a lot in the new tests. Can we extract it into a shared method?

For example, `createTestModel` can define `persist` observer that can save `ctx.data` in `TestModel.lastPersistedData` property.

```js
// in the helper method
var TestModel = ds.createModel('TestModel', {value: String}, modelSettings);
TestModel.observe('persist', function(ctx, next) {
   ctx.Model.lastPersistedData = ctx.data;
   next();
});
return TestModel;

// in the test
should(TestModel.lastPersistedData).eql(updateData);
```